nncf/torch/quantization/weights_compression.py

```diff
@@ -0,0 +1,90 @@
+                input.weight, self.scale, self.zero_point, axis, 0, 255
+            )
+
+
+def _insert_pre_compression_operations(
+    module: nn.Module, allowed_types: Dict, compress_weights=False
+) -> Optional[nn.Module]:
+    """
+    Insets weights compression with dequantization or quantization pre operation for Linear and Embedding layers.
+
+    :param module: The module to insert the weights compression.
+    :param compress_weights: Enables real compression of weights in Linear and Embedding layers.
+        If False inserts pytorch torch.fake_quantize_per_channel_affine(),
+        else compress weights to int8 and inserts custom dequantization.
+    :param allowed_types: dist with pairs (allowed type for weights compression, dimention for scale and zero point)
+    :return: The module with inserted operations. The module is not trainable if compress_weights is True.
+    """
+    for _, layer in module.named_children():
+        if not type(layer) in allowed_types:
+            _insert_pre_compression_operations(layer, allowed_types, compress_weights)
+            continue
+        q_dim = allowed_types[type(layer)]
+        input_low = torch.min(layer.weight, dim=q_dim)[0].detach()
+        input_high = torch.max(layer.weight, dim=q_dim)[0].detach()
+        scale, zero_point = get_scale_zp_from_input_low_input_high(0, 255, input_low, input_high)
+
+        if compress_weights:
+            scale = scale.unsqueeze(q_dim)
+            zero_point = zero_point.unsqueeze(q_dim)
+            layer.register_pre_forward_operation(WeightsCompressor(zero_point, scale))
+
+            compressed_weight = layer.weight.data / scale + zero_point
+            compressed_weight = torch.clamp(torch.round(compressed_weight), 0, 255)
+
+            layer.weight.requires_grad = False
+            layer.weight.data = compressed_weight.type(dtype=torch.uint8)
+        else:
+            zero_point = zero_point.type(dtype=torch.int32)
+            layer.register_pre_forward_operation(WeightsCompressor(zero_point, scale))
+
+
+def insert_pre_compression_operations(module: nn.Module, compress_weights=False) -> Optional[nn.Module]:
+    """
+    Insets weights compression with dequantization or quantization pre operation for Linear and Embedding layers.
+
+    :param module: The module to insert the weights compression.
+    :param compress_weights: Enables real compression of weights in Linear and Embedding layers.
+        If False inserts pytorch torch.fake_quantize_per_channel_affine(),
+        else compress weights to int8 and inserts custom dequantization.
+    :return: The module with inserted operations. The module is not trainable if compress_weights is True.
+    """
+    user_types = list(NNCF_WRAPPED_USER_MODULES_DICT.values())
+    allowed_types = {NNCFEmbedding: 0, NNCFLinear: 1}
+
+    for user_type in user_types:
+        if torch.nn.Embedding in user_type.__mro__:
+            allowed_types[user_type] = 0
```

Comment:
`get_target_dim_for_compression` should be 0 for `NNCF_WRAPPED_USER_MODULES_DICT`, by default. 
should be possible to reuse it

Reply:
Done

Reply:
Done